### PR TITLE
fix: clean golang code for golint

### PR DESCRIPTION
**Pull request by yulng**
Signed-off-by: yulng <wei.yang@daocloud.io>

### CODE REVIEW

**Comment by aanm** [comment]
@yulng We should avoid these sort of pull requests since they take time from committers to review them and they are not providing any bug fix nor new functionality. You are more than welcome to open one single pull request that contains multiple code cleanups.

Thank you

**Comment by pchaigno** [comment]
In addition to what André said, and as noted in the other pull request (cf. https://github.com/cilium/cilium/pull/22642#issuecomment-1344242799), I would rather wait a bit before sending those kinds of cleanups. We have just branched for v1.13 and these cleanups will just create backporting conflicts.

**Comment by yulng** [comment]
> @yulng We should avoid these sort of pull requests since they take time from committers to review them and they are not providing any bug fix nor new functionality. You are more than welcome to open one single pull request that contains multiple code cleanups.
> 
> Thank you

All changes use this PR
Thanks

**Comment by yulng** [comment]
> In addition to what André said, and as noted in the other pull request (cf. [#22642 (comment)](https://github.com/cilium/cilium/pull/22642#issuecomment-1344242799)), I would rather wait a bit before sending those kinds of cleanups. We have just branched for v1.13 and these cleanups will just create backporting conflicts.

This is the last time i clean code and all changes use this PR
Thanks

**Comment by yulng** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

It's a pity

**Review by jrajahalme — APPROVED** [review]

**Review by NikAleksandrov — APPROVED** [review]

File: pkg/bpf/bpf_linux.go [hunk 1]
```diff
@@ -595,7 +595,10 @@ recreate:
 			flags,
 		)
 		if redo {
-			ObjClose(fd)
+			err := ObjClose(fd)
+			if err != nil {
+				return fd, false, err
+			}
```
Comment (aanm) [inline pkg/bpf/bpf_linux.go]: @pchaigno @NikAleksandrov I don't know if this change was here when you left a review but does this look good to you?

Comment (qmonnet) [reply pkg/bpf/bpf_linux.go]: It probably doesn't matter if we fail to close the fd at this stage, ignoring the error (`_ = ObjClose(fd)`) would seem fine to me.

**Review by qmonnet — APPROVED** [review]
Looks good, apart from the item raised by André

File: pkg/bpf/bpf_linux.go (hunk 1 above)
Comment (yulng) [reply pkg/bpf/bpf_linux.go]: Thanks

**Comment by yulng** [comment]
@aanm André
Thanks

File: pkg/bpf/bpf_linux.go (hunk 1 above)
Comment (aanm) [reply pkg/bpf/bpf_linux.go]: @yulng I believe Quentin was suggesting to keep this code as it was before.

Comment (NikAleksandrov) [reply pkg/bpf/bpf_linux.go]: I agree with Quentin, I think we can ignore it here.

**Comment by yulng** [comment]
@aanm André
Thanks😄😄

File: pkg/bpf/bpf_linux.go (hunk 1 above)
Comment (aanm) [reply pkg/bpf/bpf_linux.go]: @yulng can we revert this change?

**Comment by pchaigno** [comment]
@yulng Please see the unresolved comment from André above.

File: pkg/bpf/bpf_linux.go (hunk 1 above)
Comment (yulng) [reply pkg/bpf/bpf_linux.go]: ok,done
Thanks

**Comment by yulng** [comment]
> @yulng Please see the unresolved comment from André above.
Just changed

**Review by aanm — APPROVED** [review]

**Review by NikAleksandrov — APPROVED** [review]

**Review by thorn3r — APPROVED** [review]

**Comment by joestringer** [comment]
Thanks for the PR, I've merged it.

@yulng did you use some tool to detect these somehow? Maybe we can integrate that tooling into the linters in the tree so that we can automatically format the code like this for all future submissions.

**Pull request merged**